GPU/GPUCommon.cpp

```diff
@@ -1808,6 +1808,70 @@ void GPUCommon::Execute_MorphWeight(u32 op, u32 diff) {
 	gstate_c.morphWeights[(op >> 24) - GE_CMD_MORPHWEIGHT0] = getFloat24(op);
 }
 
+void GPUCommon::Execute_ImmVertexAlphaPrim(u32 op, u32 diff) {
+	// Safety check.
+	if (immCount_ >= MAX_IMMBUFFER_SIZE)
+		return;
+
+	uint32_t data = op & 0xFFFFFF;
+	TransformedVertex &v = immBuffer_[immCount_++];
+
+	// Formula deduced from ThrillVille's clear.
+	int offsetX = gstate.getOffsetX16();
+	int offsetY = gstate.getOffsetY16();
+	v.x = ((gstate.imm_vscx & 0xFFFFFF) - offsetX) / 16.0f;
+	v.y = ((gstate.imm_vscy & 0xFFFFFF) - offsetY) / 16.0f;
+	v.z = gstate.imm_vscz & 0xFFFF;
+	v.u = 0.0f;  // we have no information about the scale here
+	v.v = 0.0f;  // we have no information about the scale here
+	v.w = 0.0f;  // we have no information about the scale here
+	v.color0_32 = (gstate.imm_cv & 0xFFFFFF) | (gstate.imm_ap << 24);
+	v.fog = 0.0f; // we have no information about the scale here
+	v.color1_32 = gstate.imm_scv & 0xFFFFFF;
+	int prim = (op >> 8) & 0xF;
+	if (prim != 7) {
+		immPrim_ = (GEPrimitiveType)prim;
+	} else if (prim == 7 && immCount_ == 2) {
```

Comment:
Oh, also, would be nice to use `GE_PRIM_KEEP_PREVIOUS` instead of raw magic numbers...

-[Unknown]

Reply:
Right, wasn't sure at first this was the same enum but it obviously is.